src/cascadia/TerminalSettingsEditor/Profiles.xaml

```diff
@@ -104,28 +110,31 @@ the MIT License. See LICENSE in the project root for license information. -->
                                       x:Name="StartingDirectoryUseParentCheckbox"
                                       IsChecked="{x:Bind State.Profile.UseParentProcessDirectory, Mode=TwoWay}"/>
                             </StackPanel>
-                        </ContentPresenter>
+                        </local:SettingContainer>
 
                         <!--Icon-->
-                        <ContentPresenter Style="{StaticResource SettingContainerStyle}">
+                        <local:SettingContainer x:Uid="Profile_Icon"
+                                                x:Name="IconContainer"
+                                                HasSettingValue="{x:Bind State.Profile.HasIcon, Mode=OneWay}"
+                                                ClearSettingValue="{x:Bind State.Profile.ClearIcon}">
                             <StackPanel Orientation="Horizontal">
-                                <TextBox x:Uid="Profile_Icon"
-                                         x:Name="Icon"
+                                <TextBox x:Name="Icon"

```

Comment:
does this need to be named any more?

Reply:
oh, no, we got rid of the need to name things w/ browse buttons by having a good viewmodel

Reply:
Huh, good catch. I'll take a look at the other browse button settings too.